ocaml/xapi/cli_operations.ml

```diff
@@ -2363,6 +2379,17 @@ Pool. Please provide an sr-name-label or sr-uuid parameter." in
 		Client.VM.set_name_description rpc session_id new_vm description;
 		Client.VM.set_suspend_SR rpc session_id new_vm suspend_sr_ref;
 		rewrite_provisioning_xml rpc session_id new_vm sr_uuid;
+		let recommendations = Client.VM.get_recommendations rpc session_id template in
+		let licerr = Api_errors.Server_error(Api_errors.license_restriction, [Features.name_of_feature Features.PCI_device_for_auto_update]) in
+		let want_dev = is_recommended recommendations "has-vendor-device" in
+		(
+			try Client.VM.set_has_vendor_device rpc session_id new_vm want_dev
+			with e when e = licerr ->
+				let msg = Printf.sprintf "Licence forbids setting recommended has-vendor-device for VM %s (%s)."
+					(Client.VM.get_uuid rpc session_id new_vm) (Client.VM.get_name_label rpc session_id new_vm) in
+				warn "%s" msg;
+				Cli_printer.PStderr (msg^"\n") |> printer
```

Comment:
I disagree: we don't want the `\n` for the call to `warn` (for the logfile), only when we write the message to stderr.